medcat/utils/meta_cat/data_utils.py

```diff
@@ -180,9 +186,38 @@ def encode_category_values(data: Dict, existing_category_value2id: Optional[Dict
         category_value2id = {}
 
     category_values = set([x[2] for x in data])
-    for c in category_values:
-        if c not in category_value2id:
-            category_value2id[c] = len(category_value2id)
+
+    # If categoryvalue2id is pre-defined, then making sure it is same as the labels found in the data
+    if len(category_value2id) != 0:
+        if set(category_value2id.keys()) != category_values:
+            # if categoryvalue2id doesn't match the labels in the data, then 'class_name_map' has to be defined to check for variations
+            if len(class_name_map) != 0:
+                updated_category_value2id = {}
+                for _class in category_value2id.keys():
+                    if _class in category_values:
+                        updated_category_value2id[_class] = category_value2id[_class]
+                    else:
+                        found_in = [sub_map for sub_map in class_name_map if _class in sub_map][0]
```

Comment:
Here as well, we take the first matching "map" that has a matching class in it.
If there isn't a matching "sub map" then this will fail with an `IndexError`.
What if there are more than 1? To me that sounds like a configuration mismatch and should lead to an exception.